bin/release

```diff
@@ -0,0 +1,375 @@
+         GitHub API token
+
+COMMANDS:
+  prepare: prepare services atomic updates
+  push   : atomically push updates to a remote branch
+  tag    : tag every commit corresponding to a relase & publish it
+"
+
+    exit "${exit_code}"
+}
+
+# --- Utils ---
+
+# log: log a message given a level
+#
+# usage: log LEVEL MESSAGE
+#
+#   LEVEL  : a key word in error, info, success, warning
+#   MESSAGE: message to log
+function log() {
+
+    local level="${1}"
+    local message="${2}"
+    local color
+
+    case "${level}" in
+        "error") color="${COLOR_ERROR}";;
+        "info") color="${COLOR_INFO}";;
+        "success") color="${COLOR_SUCCESS}";;
+        "warning") color="${COLOR_WARNING}";;
+        *) color="${COLOR_DEFAULT}";;
+    esac
+
+    echo -e "${color}$(printf "%-7s" "${level}" | tr "[:lower:]" "[:upper:]") ${message}${COLOR_RESET}" 1>&2
+}
+
+# --- Core ---
+
+# _update_local: sync master branch & tags
+#
+# usage: _update_local
+function _update_local() {
+
+    local release_branch
+
+    log info "☁ Fetching changes from origin"
+    git fetch origin
+    git checkout master
+    git pull origin master
+}
+
+# _bootstrap: bootstrap a new releases collection
+#
+# usage: _bootstrap
+function _bootstrap(){
+
+    release_branch="prepare-releases-${RANDOM}"
+    log info "🚉 Checking out new branch ${release_branch}"
+    git checkout -b "${release_branch}"
+}
+
+# _get_project_releases_cache_file_name: get the name of the file used as a
+# cache for a GitHub API response (JSON)
+#
+# usage: _get_project_releases_cache_file_name PROJECT
+#
+#   PROJECT: the project name
+function _get_project_releases_cache_file_name(){
+
+    local project="${1:-${GH_LRS_PROJECT}}"
+    echo "${WRK_DIR}/${project}.json"
+}
+
+# _get_project_releases_headers_file_name: get the name of the file used as a
+# cache for a GitHub API response headers
+#
+# usage: _get_project_releases_headers_file_name PROJECT
+#
+#   PROJECT: the project name
+function _get_project_releases_headers_file_name(){
+
+    local project="${1:-${GH_LRS_PROJECT}}"
+    echo "${WRK_DIR}/${project}.headers"
+}
+
+# _github_api_get_request: perform a HTTP/GET request to GitHub's API
+#
+# usage: _github_api_get_request PROJECT HEADERS CACHE ENDPOINT
+#
+#   PROJECT : the project name
+#   HEADERS : headers cache file name
+#   CACHE   : query cache file name
+#   ENDPOINT: API endpoint to query
+#
+# Nota bene: we force pagination size to 100 items for not having to deal with
+# pagination at all (it should be sufficient in most cases).
+function _github_api_get_request(){
+
+    local project="${1}"
+    local headers="${2}"
+    local cache="${3}"
+    local endpoint=${4:-releases}
+    local cmd
+
+    # Check that curl is installed and available
+    curl="$(command -v curl)"
+    if [[ -z "${curl}" ]]; then
+        log error "❌ curl is not installed (or not in your \$PATH)"
+        exit 10
+    fi
+
+    cmd="${curl}"
+
+    if [[ ! -z "${GH_TOKEN}" ]]; then
```

Comment:
This mean that it is not mandatory to have a token ?

Reply:
Yup.